### ISSUE TITLE
Image Likes

### ISSUE THREAD

**Issue by SPH73** [PacktPublishing/Django-3-by-Example#44]
I have found that when a user without a profile photo likes an image the following error occurs "photo attribute has no file associated with it" 

Anyone else?

**Comment by SheraramPrajapat1998** [comment]
I had the same error. I solved it by using a default pic in ` model ` ` photo = models.ImageField(upload_to='users/%Y/%m/%d', default='default.png', blank=True)`. This way whenever a User object(user) is created it will automatically set profile pic to default.

**Comment by SPH73** [comment]
> I had the same error. I solved it by using a default pic in `Profile` model ` photo = models.ImageField(upload_to='users/%Y/%m/%d', default='default.png', blank=True)`. This way whenever a User object(user) is created it will automatically set profile pic to default.

I did the same to resolve this and I would suggest the author could improve the book by including something as simple as this especially since the photo is not required.

**Comment by SheraramPrajapat1998** [comment]
> > I had the same error. I solved it by using a default pic in `Profile` model ` photo = models.ImageField(upload_to='users/%Y/%m/%d', default='default.png', blank=True)`. This way whenever a User object(user) is created it will automatically set profile pic to default.
> 
> I did the same to resolve this and I would suggest the author could improve the book by including something as simple as this especially since the photo is not required.



> > I had the same error. I solved it by using a default pic in `Profile` model ` photo = models.ImageField(upload_to='users/%Y/%m/%d', default='default.png', blank=True)`. This way whenever a User object(user) is created it will automatically set profile pic to default.
> 
> I did the same to resolve this and I would suggest the author could improve the book by including something as simple as this especially since the photo is not required.

Yes, I agree. It was an irritating thing until I learned about how to put default images and working with signals.

**Comment by timotif** [comment]
Thank you! This works. The default.png should be placed in the folder stated in MEDIA_URL in the settings (/media/ if you follow the book). And of course after editing the Profile class you need to make a migration to update the db.
What would be a way to retrospectively have the already registered user without profile picture pointing at the default.png?

**Comment by timotif** [comment]
I did it from the shell. Not sure it's the best way but here it is.
In the terminal run 
`python manage.py shell`

In the shell:
`from account.models import Profile`
# I want to select all the profiles with blank photo field. It's blank field, not a null, so I just compared against an empty string. Maybe not very elegant?
`blank_pic = Profile.objects.filter(photo="")`
# Iterate through these elements and assign default.png as photo
`for pic in blank_pic:
        pic.photo='default.png'
        pic.save()`
